4.queueRange.js

```diff
@@ -29,16 +29,33 @@
 //     return this.array.shift()
 //   }
 // }
-// 
+//
 // Imagine the queue contains -145, 45, 11, 98
 // range(q) should return 243
 
 // Write your function here
-const range = (queue) => {
-
-}
-
-
 
+//if empty, return
+//set diff variable
+//set max and min
+//while stack is not empty
+//get max
+//get min
+//subtract max and min to find diff
+//return diff
+const range = (queue) => {
+  if (queue.isEmpty()) return;
+  let difference = 0;
+  let max = queue.peek();
+  let min = 0
+  while (!queue.isEmpty()) {
+    max = Math.max(max, queue.dequeue());
+    min = Math.min(min, queue.dequeue());
+    // console.log(max, "max")
```

Comment:
this dequeus twice. You only want to dequeue once per iteration, so save that as a variable that you can use.